malariagen_data/ag3.py

```diff
@@ -314,6 +320,482 @@ def _repr_html_(self):
         """
         return html
 
+    def genome_sequence(self, region, inline_array=True, chunks="native"):
+        """Access the reference genome sequence.
+
+        Parameters
+        ----------
+        region: str or list of str or Region or list of Region
+            Contig name (e.g., "2L"), gene name (e.g., "AGAP007280"), genomic
+            region defined with coordinates (e.g., "2L:44989425-44998059") or a
+            named tuple with genomic location `Region(contig, start, end)`.
+            Multiple values can be provided as a list, in which case data will
+            be concatenated, e.g., ["3R", "3L"].
+        inline_array : bool, optional
+            Passed through to dask.array.from_array().
+        chunks : str, optional
+            If 'auto' let dask decide chunk size. If 'native' use native zarr
+            chunks. Also, can be a target size, e.g., '200 MiB'.
+
+        Returns
+        -------
+        d : dask.array.Array
+            An array of nucleotides giving the reference genome sequence for the
+            given contig.
+
+        """
+        region = self.resolve_region(region)
+
+        if region.contig in self.contigs:
+            d = super().genome_sequence(
+                region, inline_array=inline_array, chunks=chunks
+            )
+            return d
+
+        elif region.contig in ["2RL", "3RL"]:
+            genome = self.open_genome()
+            contigs = (
+                region.contig[0] + region.contig[1],
+                region.contig[0] + region.contig[1],
+            )
+
+            d = da.concatenate(
+                [
+                    da_from_zarr(
+                        genome[contig], inline_array=inline_array, chunks=chunks
+                    )
+                    for contig in contigs
+                ]
+            )
+
+        if region.start:
+            slice_start = region.start - 1
+        else:
+            slice_start = None
+        if region.end:
+            slice_stop = region.end
+        else:
+            slice_stop = None
+        loc_region = slice(slice_start, slice_stop)
```

Comment:
Also just noting, this code is duplicated from the superclass. Again, would be good to find a way to do this without any code duplciation.

Reply:
Yeah, i think I need to split and convert the coordinates instead like 2RL:50,000,000-70,000,000 into 2R:50,000,000-max , 2L:0-xxx

Reply:
I was trying to avoid this before, but it should simplify the code.